Libraries/Lists/VirtualizedList.js

```diff
@@ -1559,7 +1612,10 @@ class VirtualizedList extends React.PureComponent<Props, State> {
       onStartReached &&
       isWithinStartThreshold &&
       shouldExecuteNewCallback &&
-      this.state.first === 0
+      this.state.first === 0 &&
+      // On initial mount when using initialScrollIndex the offset will be 0 initially
+      // and will trigger an unexpected onStartReached.
+      (!initialScrollIndex || this._hasInteracted)
```

Comment:
This change should probably go in the onStartReached PR. This avoids triggering onStartReached when passing initialScrollIndex since our initial offset will be invalid.